bin/cml.js

```diff
@@ -29,59 +61,52 @@ const configureLogger = (level) => {
   });
 };
 
+const setupTelemetry = async (opts) => {
+  const { cml, cmlCommand: action } = opts;
+  opts.telemetryEvent = await jitsuEventPayload({ action, cml });
+};
+
 const runPlugin = async ({ $0: executable, command }) => {
-  try {
-    if (command === undefined) throw new Error('no command');
-    const path = which.sync(`${basename(executable)}-${command}`);
-    const parameters = process.argv.slice(process.argv.indexOf(command) + 1); // HACK
-    process.exit(await pseudoexec(path, parameters));
-  } catch (error) {
-    yargs.showHelp();
-    winston.debug(error);
-  }
+  if (command === undefined) throw new Error('no command');
+  const path = which.sync(`${basename(executable)}-${command}`);
+  const parameters = process.argv.slice(process.argv.indexOf(command) + 1); // HACK
+  await pseudoexec(path, parameters);
 };
 
-const handleError = (message, error) => {
+const handleError = async (message, error, yargs) => {
   if (error) {
-    winston.error(error);
-  } else {
-    yargs.showHelp();
-    console.error('\n' + message);
+    const { telemetryEvent, cml } = OPTS;
+    const event = { ...telemetryEvent, error: error.message };
+    await cml.telemetrySend({ event });
+    return;
   }
-  process.exit(1);
-};
 
-const options = {
-  log: {
-    type: 'string',
-    description: 'Maximum log level',
-    coerce: (value) => configureLogger(value) && value,
-    choices: ['error', 'warn', 'info', 'debug'],
-    default: 'info'
-  }
+  yargs.showHelp();
+  console.error('\n' + message);
 };
 
-const legacyEnvironmentVariables = {
-  TB_CREDENTIALS: 'CML_TENSORBOARD_DEV_CREDENTIALS',
-  DOCKER_MACHINE: 'CML_RUNNER_DOCKER_MACHINE',
-  RUNNER_IDLE_TIMEOUT: 'CML_RUNNER_IDLE_TIMEOUT',
-  RUNNER_LABELS: 'CML_RUNNER_LABELS',
-  RUNNER_SINGLE: 'CML_RUNNER_SINGLE',
-  RUNNER_REUSE: 'CML_RUNNER_REUSE',
-  RUNNER_NO_RETRY: 'CML_RUNNER_NO_RETRY',
-  RUNNER_DRIVER: 'CML_RUNNER_DRIVER',
-  RUNNER_REPO: 'CML_RUNNER_REPO',
-  RUNNER_PATH: 'CML_RUNNER_PATH'
-};
+process.on('uncaughtException', (err) => {
+  handleError('', err, yargs);
+});
```

Comment:
While it works as intended without this modification, I found it slightly mind-bending. Wouldn't it be better to explicitly add `async` and `await`?

```suggestion
process.on('uncaughtException', async (err) => {
  await handleError('', err, yargs);
});
```

Reply:
No simply just because it will not do anything at all

Reply:
Is it obvious what it does without including the `async` and the `await` for a casual reader?

Reply:
I'm not a JavaScript expert, but I struggle to reason an `async` function not being explicitly `await`ed and still acting like if it was. :thin

Reply:
I can add it but definitely that would be the case for the `casual` reader and not the `js dev` reader. 
Probably like the things I ask you in TPI like explicit returns... Not very golantic way